### PR TITLE
fix: handle sasjs server response

### PR DESCRIPTION
**Pull request by allanbowe**
## Issue

Link any related issue(s) in this section.

## Intent

What this PR intends to achieve.

## Implementation

What code changes have been made to achieve the intent.

## Checks

No PR (that involves a non-trivial code change) should be merged, unless all items below are confirmed!  If an urgent fix is needed - use a tar file.

- [x] All `sasjs-cli` unit tests are passing (`npm test`).
Mocked:
image
Server:
image
- [x] All `sasjs-tests` are passing (instructions available [here](https://github.com/sasjs/adapter/blob/master/sasjs-tests/README.md)).
- [x] [Data Controller](https://datacontroller.io) builds and is functional on both SAS 9 and Viya

## Testing Ubuntu
Viya
sasjs-tests-viya

Sas9
sasjs-tests-sas9

### CODE REVIEW

File: src/job-execution/WebJobExecutor.ts [hunk 1]
```diff
@@ -144,10 +145,23 @@ export class WebJobExecutor extends BaseJobExecutor {
...
-
           let jsonResponse = res.result
 
+          if (this.serverType === ServerType.Sasjs && config.debug) {
+            this.requestClient!.appendRequest(
+              jsonResponse,
+              sasJob,
+              config.debug
+            )
+            jsonResponse = JSON.parse(parseWeboutResponse(jsonResponse._webout))
```
Comment (YuryShkoda) [inline src/job-execution/WebJobExecutor.ts:156]: `JSON.parse` throws an exception.
And shouldn't `parseWeboutResponse` function return a parsed value as the name suggests?

File: src/job-execution/WebJobExecutor.ts [hunk 2]
```diff
@@ -144,10 +145,23 @@ export class WebJobExecutor extends BaseJobExecutor {
...
+          } else {
+            this.requestClient!.appendRequest(res, sasJob, config.debug)
+          }
+
+          if (this.serverType === ServerType.Sasjs) {
+            jsonResponse = JSON.parse(jsonResponse._webout)
```
Comment (YuryShkoda) [inline src/job-execution/WebJobExecutor.ts:162]: JSON.parse throws an exception.

**Review by YuryShkoda — CHANGES_REQUESTED** [review]
see comments

File: src/auth/AuthManager.ts [hunk 3]
```diff
@@ -218,19 +220,19 @@ export class AuthManager {
     return await this.getLoginForm(formResponse)
   }
 
-  private async fetchUserName(): Promise<{
+  private async fetchUserName(accessToken?: string): Promise<{
     isLoggedIn: boolean
     userName: string
   }> {
-    //For VIYA we will send request on API endpoint. Which is faster then pinging SASJobExecution.
-    //For SAS9 we will send request on SASStoredProcess
```
Comment (allanbowe) [inline src/auth/AuthManager.ts:226]: why are we removing these comments?

File: src/SASjs.ts [hunk 4]
```diff
@@ -529,8 +530,8 @@ export default class SASjs {
    * Checks whether a session is active, or login is required.
    * @returns - a promise which resolves with an object containing two values - a boolean `isLoggedIn`, and a string `userName`.
    */
-  public async checkSession() {
-    return this.authManager!.checkSession()
+  public async checkSession(accessToken?: string) {
```
Comment (allanbowe) [inline src/SASjs.ts:533]: is it worth making a comment that the access token is only used in SASjs server type?

File: src/job-execution/WebJobExecutor.ts [hunk 5]
```diff
@@ -142,12 +144,24 @@ export class WebJobExecutor extends BaseJobExecutor {
     }
 
     const requestPromise = new Promise((resolve, reject) => {
-      this.requestClient!.post(apiUrl, formData, undefined)
+      this.requestClient!.post(apiUrl, formData, authConfig?.access_token)
```
Comment (allanbowe) [inline src/job-execution/WebJobExecutor.ts:147]: again, I wonder if we need a comment to explain that the access_token is only relevant in SASjs (not Viya, for instance)

File: src/auth/AuthManager.ts (hunk 3 above)
Comment (saadjutt01) [reply src/auth/AuthManager.ts:226]: These were added by me, when changed the url for fetching username for `SASVIYA`.
It's obvious from following lines which URLs are used for which server.

File: src/job-execution/WebJobExecutor.ts (hunk 5 above)
Comment (saadjutt01) [reply src/job-execution/WebJobExecutor.ts:148]: added

File: src/SASjs.ts (hunk 4 above)
Comment (saadjutt01) [reply src/SASjs.ts:534]: Updated TSDoc

File: src/auth/AuthManager.ts (hunk 3 above)
Comment (allanbowe) [reply src/auth/AuthManager.ts:226]: ok

**Review by saadjutt01 — APPROVED** [review]

**Review by VladislavParhomchik — APPROVED** [review]

**Pull request merged**